tests/server/test_plugins.py

```diff
@@ -486,3 +491,464 @@ def test_history_of_collect_new_result(self):
+        assert expected in exc_info.value
+
+    def test_validate_component_result_should_have_status(self):
+        component_results = {
+            'Unit tests': {
+                'messages': {
+                    'info': ['GoodData'],
+                    'warn': [],
+                    'error': []
+                }
+            }
+        }
+        expected = 'Component Unit tests doesn\'t have status'
+        result = server_plugins.Result()
+        result.result = copy.deepcopy(self.result_to_validate)
+        result.result['componentResults'] = component_results
+        with pytest.raises(smoker_exceptions.ValidationError) as exc_info:
+            result.validate()
+        assert expected in exc_info.value
+
+    def test_validate_action_result_should_be_dict(self):
+        expected = 'Action result must be dictionary'
+        result = server_plugins.Result()
+        result.result = copy.deepcopy(self.result_to_validate)
+        result.result['action'] = str()
+        with pytest.raises(smoker_exceptions.ValidationError) as exc_info:
+            result.validate()
+        assert expected in exc_info.value
+
+    def test_validate_action_result_should_have_message(self):
+        action_result = {
+            'status': 'OK',
+        }
+        expected = 'Action doesn\'t have message'
+        result = server_plugins.Result()
+        result.result = copy.deepcopy(self.result_to_validate)
+        result.result['action'] = action_result
+        with pytest.raises(smoker_exceptions.ValidationError) as exc_info:
+            result.validate()
+        assert expected in exc_info.value
+
+    def test_validate_action_result_should_have_status(self):
+        action_result = {
+            'messages': {
+                'info': [],
+                'warn': [],
+                'error': []
+            }
+        }
+        expected = 'Action doesn\'t have status'
+        result = server_plugins.Result()
+        result.result = copy.deepcopy(self.result_to_validate)
+        result.result['action'] = action_result
+        with pytest.raises(smoker_exceptions.ValidationError) as exc_info:
+            result.validate()
+        assert expected in exc_info.value
+
+    def test_set_result(self):
+        result_to_validate = copy.deepcopy(self.result_to_validate)
+        result = server_plugins.Result()
+        result.set_result(result_to_validate, validate=True)
+        assert result.validated
+
+    def test_get_result(self):
+        result = server_plugins.Result()
+        result.result = copy.deepcopy(self.result_to_validate)
+        assert not result.validated
+        result.get_result()
+        assert result.validated
+
+
+def get_process_list():
+    procs = dict()
+    for proc in psutil.process_iter():
+        try:
+            pinfo = proc.as_dict(attrs=['pid', 'cmdline'])
+            procs[pinfo['pid']] = pinfo['cmdline'][0]
+        except (psutil.NoSuchProcess, IndexError, TypeError):
```

Comment:
if 'cmdline' is Null, psutil will return :
 + 'None' on Mac
 + An emty list on linux